### PR TITLE
Integrate bull contracts and leverage rebalance

### PR DESCRIPTION
**Pull request by KMKoushik**
Fixes ENG-1006

### CODE REVIEW

File: hooks/useQuoter.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Oracle, Quoter } from '../types/contracts'
 import oracleAbi from '../abis/oracle.json'
 import { QUOTER_CONTRACT } from '../constants/contracts'
 
-const useQuoter = () => {
+export const useQuoter = () => {
```
Comment (alpinechicken) [inline hooks/useQuoter.ts:8]: Why don't we just use useContractReads for the quoter?

**Review by alpinechicken — APPROVED** [review]
lgtm

File: hooks/useQuoter.ts (hunk 1 above)
Comment (KMKoushik) [reply hooks/useQuoter.ts:8]: Depends on where we use quoter. If it's inside a react component or another hook we can user useContractReads. 
But if we are using it outside react (plain ts util files) you can't use hooks there!

Comment (alpinechicken) [reply hooks/useQuoter.ts:8]: ah I see!

**Pull request merged**